### DISCUSSION TITLE
Concerns with API key in the frond end code

### DISCUSSION THREAD

**Discussion by keenan91** [wevm/wagmi#1163] [Q&A]
Hello I've been loving using the WAGMI hooks but should I be concerned about exposing an Alchemy or infura API key when using the provider in a Nextjs app

**Comment by tmm** [comment]
See https://github.com/wagmi-dev/wagmi/discussions/232#discussioncomment-2297454

**Reply by keenan91** [reply]
Very nice would this also apply to calls to alchemy API's
